.github/workflows/deploy.yml

```diff
@@ -39,10 +39,19 @@ jobs:
       if: github.ref == 'refs/heads/beta'
       run: echo "BETA=true" >> $GITHUB_ENV
 
-    - name: Use scripts and templates from master branch
+    - name: Use deploy script from master branch
       run: |
         git fetch --no-tags --prune --depth=1 origin master
-        git checkout origin/master -- .github/deploy.sh util/gh-pages/ util/*.py
+        git checkout origin/master -- .github/deploy.sh
```

Comment:
This is probably what you meant with:

> This removes the template files and the scripts used for deployment from the checkout. This was added in #5517. I don't think we ever needed those there. Not sure though.

We still need the content of `util/gh-pages/` as it contains the index file and the `versions.py` to export the version overview. 

The other python files are no longer needed but still referenced in the deploy script itself. You probably need to update that script in addition to your changes :)

Reply:
The contents of `util/gh-pages` is also on the `beta` branch and on every tagged commit. So checking out the master `index.html` file would even break the deployment for beta and stable, since it assumes that there are information in the `lints.json` file that aren't there yet on beta and stable.

Good point on the `versions.py` script. I don't _think_ we have to check it out, but I'll double check.

The `deploy.sh` script is something I have to check on. It should be written in a way, that it doesn't assume anything of the Clippy repo structure, except the necessary files for deployment. That way, if a bug occurs in the deploy script, it can be fixed on master and this fix will be immediately available on beta and stable. So that the python files are called from the deploy script is kind of a problem. I'll see how to best handle that.

Reply:
> Good point on the `versions.py` script. I don't think we have to check it out, but I'll double check.

Please check that. I also thought about rewriting the version collection in our dev tool and implementing #6492 along with it. But that's pretty low on my to-do list right now. :upside_down_face:

Reply:
I don't think RIIRing the `versions.py` script makes sense :smile:

Reply:
I mainly thought about doing it for adding the lint implementation version to the website (somehow...). But a RIIR is not necessary. I agree with you :upside_down_face: